java/src/main/java/ai/onnxruntime/OrtSession.java

```diff
@@ -1339,6 +1339,19 @@ public void addCoreML(Map<String, String> providerOptions) throws OrtException {
       addExecutionProvider(CoreMLProviderName, providerOptions);
     }
 
+    /**
+     * Adds WebGPU as an execution backend.
+     *
+     * This method does not accept any parameters. The `providerOptions` is set internally
+     * to an empty map for the WebGPU execution provider configuration.
```

Comment:
interesting that it doesn't use provider options. is it planned for the WebGPU EP to not use provider options in the future too?

Reply:
It does take provider config: https://github.com/microsoft/onnxruntime/blob/56f101839603f2c875234b8baca4dc0cf5f8f973/onnxruntime/core/providers/webgpu/webgpu_execution_provider.cc#L753

@fs-eire does the usage look okay to you?

Reply:
here, it doesn't pass in the provider options to the WebGPU provider factory, but only the session options config values:

https://github.com/microsoft/onnxruntime/blob/11fda2adb93a48404fa29e090f651b5825f95f0e/onnxruntime/core/session/provider_registration.cc#L137

Reply:
oh, I see that provider options are also inserted into config_options.

https://github.com/microsoft/onnxruntime/blob/11fda2adb93a48404fa29e090f651b5825f95f0e/onnxruntime/core/session/provider_registration.cc#L93-L95

given that, it might be good to also support passing in a `providerOptions` map in this Java API.

Reply:
I don't know how JAVA API works, but what WebGPU EP uses via C API is this one:

```c++
  ORT_API2_STATUS(SessionOptionsAppendExecutionProvider, _In_ OrtSessionOptions* options,
                  _In_ const char* provider_name,
                  _In_reads_(num_keys) const char* const* provider_options_keys,
                  _In_reads_(num_keys) const char* const* provider_options_values,
                  _In_ size_t num_keys);
```

The provider options are key-value pairs that defined in:
https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/webgpu/webgpu_provider_options.h (keys are passed without "WebGPU:" prefix)

Reply:
The Java API flattens the provider options map into two arrays and calls the C API `SessionOptionsAppendExecutionProvider`. If there are provider flags for the WebGPU EP then it should accept a map in Java.
https://github.com/microsoft/onnxruntime/blob/main/java/src/main/java/ai/onnxruntime/OrtSession.java#L1283